### PR TITLE
fix(security): add horizontal authentication for service

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
Add horizontal authentication for service

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by smallsheeeep — APPROVED** [review]
LGTM

**Review by MarkPotato777 — DISMISSED** [review]
LGTM

File: server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/TestLoginManager.java [hunk 1]
```diff
@@ -173,6 +176,7 @@ public static boolean isOAuthTestLoginRequest(HttpServletRequest request) {
         return "test".equals(parseRegistrationName(registrationId));
     }
 
+    @PreAuthenticate(actions = "read", resourceType = "ODC_INTEGRATION", isForAll = true)
```
Comment (ungreat) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/TestLoginManager.java:179]: `loadLdapContext` is executed before login, so it does not need to be annotated with @PreAuthenticate.

**Review by LuckyPickleZZ — DISMISSED** [review]
LGTM

**Review by yizhouxw — DISMISSED** [review]

**Review by smallsheeeep — APPROVED** [review]

**Review by ungreat — APPROVED** [review]
LGTM

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]

**Review by MarkPotato777 — APPROVED** [review]

**Review by LuckyPickleZZ — APPROVED** [review]

**Pull request merged**